addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -316,66 +370,194 @@ const FontFamilyPickerUserValueWidget = SelectUserValueWidget.extend({
-        this.dialog.add(addGoogleFontDialog, {
-            title: _t("Add a Google Font"),
-            onClickSave: async (state, inputEl) => {
-                // if font page link (what is expected)
-                let m = inputEl.value.match(/\bspecimen\/([\w+]+)/);
-                if (!m) {
-                    // if embed code (so that it works anyway if the user put the embed code instead of the page link)
-                    m = inputEl.value.match(/\bfamily=([\w+]+)/);
-                    if (!m) {
-                        inputEl.classList.add('is-invalid');
-                        return;
+            get getGoogleFontList() {
+                return [{options: async (term) => {
+                    if (!this.googleFontList) {
+                        await rpc("/website/google_font_metadata").then((data) => {
+                            this.googleFontList = data.familyMetadataList.map((font) => font.family);
+                        });
                     }
-                }
-
-                let isValidFamily = false;
-
+                    const lowerCaseTerm = term.toLowerCase();
+                    const filtered = this.googleFontList.filter((value) => value.toLowerCase().includes(lowerCaseTerm));
+                    return filtered.map((fontFamilyName) => {
+                        return {
+                            label: fontFamilyName,
+                            value: fontFamilyName,
+                        };
+                    });
+                }}];
+            }
+            async onGoogleFontChange(selected) {
+                this.fileInput.el.value = "";
+                this.state.uploadedFonts = [];
+                this.state.uploadedFontName = undefined;
+                this.state.uploadedFontFaces = undefined;
                 try {
-                    // Font family is an encoded query parameter:
-                    // "Open+Sans" needs to remain "Open+Sans".
-                    const result = await fetch("https://fonts.googleapis.com/css?family=" + m[1] + ':300,300i,400,400i,700,700i', {method: 'HEAD'});
+                    const fontFamily = selected.value;
+                    const result = await fetch(`https://fonts.googleapis.com/css?family=${encodeURIComponent(fontFamily)}:300,300i,400,400i,700,700i`, {method: 'HEAD'});
                     // Google fonts server returns a 400 status code if family is not valid.
                     if (result.ok) {
-                        isValidFamily = true;
+                        const linkId = `previewFont${fontFamily}`;
+                        if (!document.querySelector(`link[id='${linkId}']`)) {
+                            const linkEl = document.createElement("link");
+                            linkEl.id = linkId;
+                            linkEl.setAttribute("href", result.url);
+                            linkEl.setAttribute("rel", "stylesheet");
+                            document.head.appendChild(linkEl);
```

Comment:
This is about being able to show the preview: within the actual page, the CSS will be reloaded and will contain the new font after confirmation. But for the preview we are in the main iframe and we will not impact the backend's CSS - since it normally never needs to be used in backend pages. Instead we just add a link to the font in the main iframe's `<head>` so that the preview can be shown inside the dialog.